helper function to print out all DDP-relevant env vars

huangyi1979 opened the pull request:
Summary:
GH issue: https://github.com/pytorch/pytorch/issues/40105

Add a helper function to DDP to print out all relevant env vars for debugging

Test Plan:
test through unittest, example output:
 ---
env:RANK=3
env:LOCAL_RANK=N/A
env:WORLD_SIZE=N/A
env:MASTER_PORT=N/A
env:MASTER_ADDR=N/A
env:CUDA_VISIBLE_DEVICES=N/A
env:GLOO_SOCKET_IFNAME=N/A
env:GLOO_DEVICE_TRANSPORT=N/A
env:NCCL_SOCKET_IFNAME=N/A
env:NCCL_BLOCKING_WAIT=N/A
...
 ---

Differential Revision: D22490486

mrshenli commented on torch/nn/parallel/distributed.py:
```diff
@@ -263,6 +320,8 @@ def __init__(self, module, device_ids=None,
                  find_unused_parameters=False,
                  check_reduction=False):
+        # Dump DDP-relevant environment variables for debugging
+        _dump_DDP_relevant_env_vars()
```
Shall we remove this from DDP ctor, as this info is probably only useful when debugging. If applications need to check env vars, they can explicitly call `_dump_DDP_relevant_env_vars`.

mrshenli reviewed (commented):
Thanks for working on this!! 

Shall we add a test to the `_DistTestBase` class in `pytorch/test/distributed/test_distributed.py`. We don't need to check all env vars in the test, checking `MASTER_ADDR`, `MASTER_PORT`, and, `WORLD_SIZE` would be sufficient, which are set here:

https://github.com/pytorch/pytorch/blob/0fc0a9308ab1e8f6b65d2565c45f54afb449d438/test/distributed/test_distributed.py#L2224-L2226

Commit:
helper function to print out all DDP-relevant env vars (#41297)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/41297

GH issue: https://github.com/pytorch/pytorch/issues/40105

Add a helper function to DDP to print out all relevant env vars for debugging

Test Plan:
test through unittest, example output:
 ---
env:RANK=3
env:LOCAL_RANK=N/A
env:WORLD_SIZE=N/A
env:MASTER_PORT=N/A
env:MASTER_ADDR=N/A
env:CUDA_VISIBLE_DEVICES=N/A
env:GLOO_SOCKET_IFNAME=N/A
env:GLOO_DEVICE_TRANSPORT=N/A
env:NCCL_SOCKET_IFNAME=N/A
env:NCCL_BLOCKING_WAIT=N/A
...
 ---

Differential Revision: D22490486

fbshipit-source-id: dd13c725f98464c8b28ae4f5b243d51f25634a40

mrshenli reviewed (approved):
LGTM! Thanks for adding this!
